packages/ts/react-crud/test/autogrid.spec.tsx

```diff
@@ -505,16 +505,52 @@ describe('@hilla/react-crud', () => {
         await assertColumns(grid, 'email', 'firstName');
       });
 
-      it('renders custom columns at the end', async () => {
-        const NameRenderer = ({ item }: { item: Person }): JSX.Element => (
+      it('renders custom columns at the specified index by visibleColumns', async () => {
+        const FullNameRenderer = ({ item }: { item: Person }): JSX.Element => (
           <span>
             {item.firstName} {item.lastName}
           </span>
         );
+        const FullNameHyphenRenderer = ({ item }: { item: Person }): JSX.Element => (
+          <span>
+            {item.firstName}-{item.lastName}
+          </span>
+        );
+        const grid = await GridController.init(
+          render(
+            <TestAutoGrid
+              visibleColumns={['fullName', 'gender', 'email', 'secondFullName', 'vip', 'birthDate', 'shiftStart']}
+              customColumns={[
+                <GridColumn key="fullName" autoWidth renderer={FullNameRenderer}></GridColumn>,
+                <GridColumn key="secondFullName" autoWidth renderer={FullNameHyphenRenderer}></GridColumn>,
+              ]}
+            />,
+          ),
+          user,
+        );
+        await assertColumns(grid, '', 'gender', 'email', '', 'vip', 'birthDate', 'shiftStart');
+        expect(grid.getBodyCellContent(0, 0)).to.have.rendered.text('Jane Love');
+        expect(grid.getBodyCellContent(0, 3)).to.have.rendered.text('Jane-Love');
+      });
+
+      it('renders custom columns at the end if visibleColumns is absent', async () => {
+        const FullNameRenderer = ({ item }: { item: Person }): JSX.Element => (
+          <span>
+            {item.firstName} {item.lastName}
+          </span>
+        );
+        const FullNameHyphenRenderer = ({ item }: { item: Person }): JSX.Element => (
+          <span>
+            {item.firstName}-{item.lastName}
+          </span>
+        );
```

Comment:
It seems these could be extracted into the upper `describe` block and then reused between the examples.